ui/src/components/Configuration/ConfigurationThreadPools.vue

```diff
@@ -0,0 +1,223 @@
+          <span v-if="!loading">Update Threadpools</span>
+        </FeatherButton>
+      </div>
+    </div>
+  </FeatherExpansionPanel>
+</template>
+
+<script setup lang="ts">
+import { computed, reactive, ref } from 'vue'
+import { useStore } from 'vuex'
+
+import { FeatherInput } from '@featherds/input'
+import { FeatherButton } from '@featherds/button'
+import { FeatherExpansionPanel } from '@featherds/expansion'
+import { FeatherChip, FeatherChipList } from '@featherds/chips'
+import { FeatherSpinner } from '@featherds/progress'
+
+import { populateProvisionD, putProvisionDService } from '@/services/configurationService'
+import { useConfigurationToast } from './hooks/configurationToast'
+import { threadPoolKeys } from './copy/threadPoolKeys'
+import { ConfigurationHelper } from './ConfigurationHelper'
+
+/**
+ * Local State
+ */
+const threadPoolsErrors = ref<Record<string, boolean>>({})
+const errorMessage = 'Thread pool values have to be between 1 and 2000.'
+const threadPoolsActive = ref(false)
+const loading = ref(false)
+
+const threadPoolData = computed(() => {
+  const localThreads: Record<string, string> = {}
+  threadPoolKeys.forEach((key) => (localThreads[key] = store?.state?.configuration?.provisionDService?.[key]))
+  return reactive(localThreads)
+})
+
+const unTouchedThreadPoolData = computed(() => {
+  const localThreads: Record<string, string> = {}
+  threadPoolKeys.forEach((key) => (localThreads[key] = store?.state?.configuration?.provisionDService?.[key]))
+  return reactive(localThreads)
+})
+
+/**
+ * Hooks
+ */
+const store = useStore()
+const { updateToast } = useConfigurationToast()
+
+/**
+ * User has opted to update threadpool data.
+ */
+const updateThreadpools = async () => {
+  loading.value = true
+  // Clear Errors
+  threadPoolsErrors.value = {}
+
+  // Set Current Threadpool state.
+  const currentThreadpoolState = threadPoolData.value
+  const updatedProvisionDData = store?.state?.configuration?.provisionDService
+
+  // Validate Threadpool Data
+  threadPoolKeys.forEach((key) => {
+    const val = parseInt(currentThreadpoolState?.[key])
+    if (val < 1 || val > 2000) {
+      threadPoolsErrors.value[key] = true
+    }
+  })
+
+  let toastMessage = {
+    basic: 'Error!',
+    detail: errorMessage,
+    hasErrors: true
+  }
+  // If there are no errors.
+  if (Object.keys(threadPoolsErrors.value).length === 0) {
+
+    try {
+      // Set Update State
+      threadPoolKeys.forEach((key) => {
+        if (updatedProvisionDData?.[key]) {
+          updatedProvisionDData[key] = parseInt(currentThreadpoolState?.[key])
+        }
+      })
+      if (updatedProvisionDData) {
+        updatedProvisionDData['requisition-def'] = ConfigurationHelper.stripOriginalIndexes(updatedProvisionDData['requisition-def'])
+      }
+      // Push Updates to Server
+      await putProvisionDService(updatedProvisionDData)
+      // Redownload + Populate Data.
+      await populateProvisionD(store)
```

Comment:
The trip here is from component => service => store => service.
Elsewhere we just call dispatch on the store directly, to make it comp => store => service. Or did you want to avoid that for better fn names?

Reply:
In my head tightly coupling a component to the store via an explicit dispatch is a bad idea from a separation of concerns standpoint, but I did remember thinking when I wrote that one-liner that it goes against the grain of the rest of the project so I might have to remove it.

So I'll leave it up to you. I personally prefer the cleaner function names, but I'm also happy to replace with the straight dispatch if you don't want to muddy up the styles.